client-device-connection-base/pom.xml

```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2024 Contributors to the Eclipse Foundation
+   
+    See the NOTICE file(s) distributed with this work for additional
+    information regarding copyright ownership.
+   
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License 2.0 which is available at
+    http://www.eclipse.org/legal/epl-2.0
+   
+    SPDX-License-Identifier: EPL-2.0
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.hono</groupId>
+    <artifactId>hono-bom</artifactId>
+    <version>2.6.0-SNAPSHOT</version>
+    <relativePath>../bom</relativePath>
+  </parent>
+  <artifactId>client-device-connection-base</artifactId>
+
+  <name>Device Connection client</name>
+  <description>Base classes for client for accessing device connection information in a remote cache / data grid.</description>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>hono-legal</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>hono-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>hono-client-common</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.opentracing</groupId>
+      <artifactId>opentracing-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.smallrye.config</groupId>
+      <artifactId>smallrye-config-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.vertx</groupId>
+      <artifactId>vertx-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.vertx</groupId>
+      <artifactId>vertx-health-check</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-core</artifactId>
+      <optional>true</optional>
```

Comment:
If this is not needed, then we should remove the dependency completely.

Reply:
This was copied from the original `client-device-connection-infinispan` module. I have no idea why it was marked optional there but I kept it in both the new `client-device-connection-base` and the `client-device-connection-infinispan` module.

I'll investigate whether this can be removed from both of them and do so if that is the case.

Reply:
So if I remove it from both modules I get a compilation error:

```
[ERROR] /Users/stefan.freyr/work/contrib/controlant-eclipse/eclipse-hono/client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/QuarkusReflectionRegistration.java:[17,38] package io.quarkus.runtime.annotations does not exist
[ERROR] /Users/stefan.freyr/work/contrib/controlant-eclipse/eclipse-hono/client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/QuarkusReflectionRegistration.java:[23,2] cannot find symbol
  symbol: class RegisterForReflection
```

So I can remove this dependency from the base module but for the -infinispan module to compile, it's still needed there.

But again, why it's marked optional is not quite clear to me and also, if it's marked optional, is it necessary to do all these exclusions?

Another question is also, should this Quarkus Reflection Registration be split up between the modules?

Here is the class that is requiring this dependency:
https://github.com/controlant-org/eclipse-hono/blob/split_command_router_modules/client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/QuarkusReflectionRegistration.java

The dependency is providing the `@RegisterForReflection` annotation.

Should the stuff defined using the `classNames` property be moved to a class in the base module while the rest (which I believe is all related to infinispan in some way) would be in the infinispan module.

Reply:
Ah, I see. The `@RegisterForReflection` annotation is needed for the Quarkus native build. It is only needed during compiling. I guess we should better mark the dependency as `provided` then as it will be pulled in by other components anyway.

And no, for the time being, we should only keep the `QuarkusReflectionRegistration` as-is in client-device-connection-infinispan. The Quarkus Redis module should already register all classes it needs for building native executables.

Reply:
So to be clear. Do you want me to change the dependency scope to `provided`?

If so, should I remove the `<optional>true</optional>`?

Also, should I remove the exclusions?

Reply:
What about this @sophokles73 ? Is my understanding here correct? Should I change this to use the provided scope and remove the optional? And keep the exclusion rules?

Reply:
> Should I change this to use the `provided` scope and remove the `optional`?

yes and yes

> And keep the `exclusion` rules?

no, they should be removed FMPOV

Reply:
allright... done and all tests seem to pass just fine.

I haven't done any tests on the native images though, I'm assuming that's done in the CI pipelines?